### ISSUE TITLE
NotImplemented should be a valid return result for a function that is annotated to return a TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3171]
The following should not return an error because NotImplemented is an instance of a class that derives from `Any`.

```python
from typing import TypeVar

T = TypeVar("T")

def foo(self, obj: T) -> T:
    return NotImplemented
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.229, which I just published. It will also be included in the next release of pylance.
